src/melee.cpp

```diff
@@ -1015,6 +1017,7 @@ matec_id player::pick_technique( Creature &t, const item &weap,
     bool downed = t.has_effect( effect_downed );
     bool stunned = t.has_effect( effect_stunned );
     bool wall_adjacent = g->m.is_wall_adjacent( pos() );
+    bool is_humanoid = ( static_cast <monster *>( &t )->type->in_species( HUMAN ) );
```

Comment:
This cast will fail badly and cause UB when `t` is not actually a monster.

Reply:
try this
```suggestion
    bool is_humanoid = t->is_player() || ( static_cast <monster *>( &t )->type->in_species( HUMAN ) );
```
If it's not a player, it's a monster.